stablehlo/dialect/StablehloOps.cpp

```diff
@@ -3591,6 +3591,59 @@ Operation* ReduceWindowOp::getReductionOp(int resultIndex) {
 // ReducePrecisionOp
 //===----------------------------------------------------------------------===//
 
+namespace {
+// Print attributes as e#m#
+void printExponentMantissa(AsmPrinter& p, IntegerAttr exponent,
+                           IntegerAttr mantissa) {
+  p << 'e';
+  p.printAttributeWithoutType(exponent);
+  p << 'm';
+  p.printAttributeWithoutType(mantissa);
+}
+
+void printExponentMantissa(AsmPrinter& p, Operation*, IntegerAttr exponent,
+                           IntegerAttr mantissa) {
+  printExponentMantissa(p, exponent, mantissa);
+}
+
+// Parse e#m# as exponent=# and mantissa=#
+ParseResult parseExponentMantissa(AsmParser& parser, IntegerAttr& exponent,
+                                  IntegerAttr& mantissa) {
+  llvm::SMLoc loc = parser.getCurrentLocation();
+  llvm::StringRef expMan;
+  if (parser.parseKeyword(&expMan)) {
+    return failure();
+  }
+
+  // Validate format e#m#
+  llvm::Regex expManRegex("^e([0-9]+)m([0-9]+)$");
+  llvm::SmallVector<llvm::StringRef> matches;
+  if (!expManRegex.match(expMan, &matches)) {
+    return parser.emitError(loc,
+                            "expected exponent mantissa in format e#m#, saw ")
+           << expMan;
+  }
+
+  // Parse off digits of exp/man
+  assert(matches.size() == 3);  // matches[0] is entire regex match.
+  llvm::StringRef expS = matches[1];
+  llvm::StringRef manS = matches[2];
+
+  // Parse as base 10 integer strings
+  int exp, mant;
+  if (expS.getAsInteger(/*radix=*/10, exp) ||
+      manS.getAsInteger(/*radix=*/10, mant)) {
+    parser.emitError(loc, "unable to parse exponent or mantissa ")
+        << expS.str() << ", " << manS.str();
+    return failure();
+  }
```

Comment:
Nit: split into two error messages? You have two tests for that anyway :)

Reply:
split into two